cli/internal/cmdutil/cmdutil.go

```diff
@@ -216,3 +221,20 @@ func (b *CmdBase) LogError(format string, args ...interface{}) {
 	b.Logger.Error("error", err)
 	b.UI.Error(fmt.Sprintf("%s%s", ui.ERROR_PREFIX, color.RedString(" %v", err)))
 }
+
+// LogWarning logs an error and outputs it to the UI.
+func (b *CmdBase) LogWarning(prefix string, err error) {
+	b.Logger.Warn(prefix, "warning", err)
+
+	if prefix != "" {
+		prefix = " " + prefix + ": "
+	}
+
+	b.UI.Error(fmt.Sprintf("%s%s%s", ui.WARNING_PREFIX, prefix, color.YellowString(" %v", err)))
```

Comment:
@mehulkar Can you help me understand why this `LogWarning` method is using this ... unorthodox ... pair?

- `b.Logger.Warn`
- `b.UI.Error`

Reply:
Sure! It's because I'm a dumdum. https://github.com/vercel/turborepo/pull/2135